docs/src/errortypes.md

```diff
@@ -0,0 +1,263 @@
+
+If you expect a `'\t'` byte to always be present, such that `findfirst` always should return an `Int` for valid input, you can add a typeassert in the function to assert that the return value of `findfirst` must be, say, an `Integer`. Then, the compiler will know that if the typeassert passes, the value returned by `findfirst` cannot be `nothing` (and hence in this case must be `Int`):
+
+```julia
+julia> function pos_after_tab(v::AbstractArray{UInt8})
+    p = findfirst(isequal(UInt8('\t')), v)::Integer
+    p + 1
+end
+
+julia> @report_call pos_after_tab(codeunits("a\tb"))
+No errors detected
+```
+
+The code will still error at runtime due to the typeassert if `findfirst` returns `nothing`, but JET will no longer detect it as an error, as violating a typeassert is not considered a type error to JET.
+Note that adding a typeassert also improves code quality:
+* The programmer's intent to never observe `nothing` is communicated clearly
+* `p` is inferred to be `Int` instead of a union, and this more precise type inference generates more efficient code.
+* More precise inference reduce the risk of invalidations from the code.
+
+A special case occurs when loading `Union`-typed fields from structs.
+Julia does not realize that loading the same field multiple times necessarily returns the same object. Hence, in the following example:
+
+```julia
+julia> struct Foo
+    x::Union{Int, Nothing}
+end
+
+julia> function f(x)
+    if x.x === nothing
+        nothing
+    else
+        x.x + 1
+    end
+end;
+
+julia> @report_call f(Foo(1))
+═════ 1 possible error found ═════
+┌ @ REPL[5]:6 x.x + 1
+│ no matching method found `+(::Nothing, ::Int64)` (1/2 union split): ((x::Foo).x::Union{Nothin g, Int64} + 1)
+└─────────────
+```
+
+We might reasonably expect the compiler to know that in the `else` branch, `x.x` must be an `Int`, since it just checked that it is not `nothing`. However, the compiler does not know that the value obtained from loading the `x` field in the expression `x.x` on the like with the if statement is the same value as the value obtained when loading the `x` field in the `x.x + 1` statement.
+You can solve this issue by assigning `x.x` to a variable:
+
+```julia
+julia> function f(x)
+    y = x.x
+    if y === nothing
+        nothing
+    else
+        y + 1
+    end
+end;
+
+julia> @report_call f(Foo(1))
+No errors detected
+```
+
+## `X` is not defined
+### Description
+This happens when a name `X` is used in a function, but no object named `X` can be found.
+
+### Example
+```julia
+julia> f(x) = foo(x) + 1;
+
+julia> @report_call f(1)
+═════ 1 possible error found ═════
+┌ @ REPL[4]:3 foo(x)
+│ `Main.foo` is not defined
+└─────────────
+```
+
+### How to fix
+This error can have a couple of causes:
+* `X` is misspelled. If so, correct the typo
+* `X` exists, but cannot be reached from the scope of the function.
+  If so, pass it in as an argument to the offending function, or mark it `const`
```

Comment:
> mark it `const`

I don't know what this refers to. It would be helpful to see an example of the solution in addition to an example of the problem.

Reply:
Sorry, this was me confusing two different solutions. I've removed it.